### ISSUE TITLE
Cuda/10.1,11.1 nightly failures in Kokkos_ArithTraits.hpp:  error: no suitable constructor exists to convert from "float" to "Kokkos::Experimental::half_t"

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos-kernels#850]
Kokkos-Kernels Cuda/10.1 and 11.1 (with c++17 enabled) nightly builds (Volta70 arch, kokkos-dev-2 testbed) began failing after Kokkos updates in PR kokkos/kokkos#3593 with error messages of the form:

```bash
02:39:46 /home/jenkins/slave/workspace/KokkosKernels_KokkosDev2_CUDA10_1/kokkos-kernels/src/Kokkos_ArithTraits.hpp(686): error: no suitable constructor exists to convert from "float" to "Kokkos::Experimental::half_t"
...
```

Though the failure follows changes in kokkos, this may indicate changes needed in kokkos-kernels that are coupled to the changes in kokkos - @crtrott @e10harvey any suggestions for resolution to the error?

Sample reproducer instructions (kokkos-dev-2):

```bash
module load sems-env sems-cmake/3.12.2 sems-cuda/10.1 sems-gcc/7.3.0

$KOKKOSKERNELS_PATH/cm_generate_makefile.bash --with-devices=Cuda,OpenMP --arch=SNB,Volta70 --compiler=$KOKKOS_PATH/bin/nvcc_wrapper --cxxflags="-O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="14" --with-cuda  --kokkos-path=$KOKKOS_PATH --kokkoskernels-path=$KOKKOSKERNELS_PATH --with-scalars='double,complex_double' --with-ordinals=int --with-offsets=int,size_t --with-layouts=LayoutLeft --no-examples
```

**Comment by ndellingwood** [comment]
Looks like PR #849 corresponds to the Kokkos changes mentioned above

**Comment by ndellingwood** [comment]
Issue due to need for PR #849 to be merged in sync with kokkos/kokkos#3593 and changes were coupled, not really a bug but we should probably add a label to track these types of changes.
